### ISSUE TITLE
ui: slide{Up,Down,Left,Right} should slide as much as possible

### ISSUE THREAD

**Issue by eaburns** [eaburns/T#239]
Right now these four functions try to slide by the given delta value. If they cannot slide by that amount, they don't move and return false. Instead, they should try to slide as much as possible up to delta. This has two benefits: 1) often if each frame slides a bit, it can amount to enough further up the recursion stack, 2) frames will be able to slide flush with the side of the window.

**Comment by skiesel** [comment]
@eaburns I have the fixes in for slide up and down -- I'm curious about how to trigger the left and right sliding in the UI?

**Comment by eaburns** [comment]
If you drop a column in a narrow space where it doesn't fit, the column into which it is being dropped will slide to the sides (first one, then the other).

**Comment by skiesel** [comment]
Hm. You'll have to create a video I think. It looks like when I drag and drop columns, all the horizontal space is used to the fullest.

**Comment by eaburns** [comment]
It should work if you drop a columns right next to the edge of another.

**Comment by eaburns** [comment]
I didn't try this out yet, but I think that it needs to recur, if that makes sense. So if I keep growing the bottom-most sheet, eventually all sheets will be pushed up so that only their tags show with no small gaps between.

**Comment by skiesel** [comment]
That is already the case :-)

**Comment by eaburns** [comment]
Very nice. Looks great.

**Comment by skiesel** [comment]
Sooooo... no testing but everything looks good -- here's the final piece for horizontal sliding:

https://youtu.be/cqsBnf-Q-Sg

**Comment by eaburns** [comment]
The sliding looks good.

What's up with the columns disappearing when you drop them? Is that Issue
#285?

**Comment by skiesel** [comment]
As you may remember from the other video where there were rendering issues, my machine gets pretty laggy when I'm doing a screen recording. I think it is #285 but it's hard to tell because the timing of mouse release and shift release get all weird and laggy.

**Comment by eaburns** [comment]
We should never lose things, even with lag.

**Comment by skiesel** [comment]
Agreed, but I am pretty sure if #285 gets fixed then this won't be an issue even with the lag.
